src/librustc_interface/queries.rs

```diff
@@ -74,6 +75,8 @@ pub struct Queries<'comp> {
     arenas: Once<AllArenas>,
     forest: Once<hir::map::Forest>,
```

Comment:
You can add `hir::map::Forest` to the list of types in `arena.rs` with a `[few]` attribute then we can remove this field and allocate it from `Arena`.